README.md

```diff
@@ -67,6 +67,69 @@ And defines the following commands, mostly a mapping to xmonad key combinations.
+> (extending the full height of the screen), and all remaining windows are
+> placed in the other pane. If either pane has more than one window, that pane
+> will be evenly split, horizontally, to show them all.
+> You can use the keyboard shortcuts above to control which window(s), and
+> how many, are in the main pane, as well as the horizontal size of the main
+> pane vs. the other pane.
+* *Tall-Right*
+> Exactly the same as *Tall*, but the main pane is on the right, with the other
+> pane on the left.
+* *Wide*
+> The rotated version of *Tall*, where the main pane is on the _top_ (extending
+> the full width of the screen), and the other pane is on the bottom.
+> If either pane has more than one window, that pane will split vertically
+> instead of horizontally.
+* *Middle-Wide*
+> A three-column layout, with one main pane in the center (extending the full
+> height of the screen) and two other panes, one on each side of the main pane.
+> Despite its name, this layout actually resembles *Tall* much more than *Wide*,
+> because panes are split horizontally if they contain more than one window.
+> You can control how many windows are in the main pane as usual; other windows
+> will be assigned as evenly as possible between the left and the right pane.
+* *Widescreen-Tall*
+> This mode is like *Tall*, but if there are multiple windows in the main pane,
+> they split the main pane vertically rather than horizontally.
+> The other pane still splits windows horizontally, like *Tall*.
+> This layout gets its name because it probably makes the most sense on very
+> wide screens, with a large main pane consisting of several columns, and all
+> remaining windows stacked into the final column.
+> Other layouts that work well on very wide screens include any that allow for
+> more than two columns (to take advantage of the screen width), such as
+> *Middle-Wide* or *Column*.
+* *Fullscreen*
+> In this layout, the currently focused window takes up the entire screen, and
+> the other windows are not visible at all.
+> You can rotate between each of the windows using the "focus the next window"
+> shortcut, as usual.
+* *Column*
+> This layout has one column per window, with each window extending the full
+> height of the screen.
+> The farthest-left window is considered the "main" window in the sense that
+> you can change its size with the "shrink/expand the main pane" shortcuts;
+> the other windows split the remaining space evenly.
+* *Row*
+> The rotated version of *Column*, where each window takes up an entire row,
+> extending the full width of the screen.
+* *Floating*
+> This mode makes all windows "floating", allowing you to move and resize them
+> as if Amethyst were temporarily deactivated.
+> Unlike the other modes, this will mean that windows can be placed "on top of"
+> each other, obscuring your view of some windows.
+* *Binary Space Partitioning (BSP)*
+> [I'm actually not sure how this one works; feel free to open a PR to explain
```

Comment:
Description of BSP:

This layout does not have a main pane in the way that other layouts do. Each pane can be split into two panes along whatever axis is longer. This is recursive such that pane A can be split in the middle into pane A on the left and pane B on the right; pane B can be split in the middle into pane B on top and pane C on bottom; pane C can be split in the middle into pane C on the left and pane D on the right; and so on.